x-pack/plugins/ai_infra/product_doc_base/server/plugin.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Path from 'path';
+import type { Logger } from '@kbn/logging';
+import { getDataPath } from '@kbn/utils';
+import type { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from '@kbn/core/server';
+import { SavedObjectsClient } from '@kbn/core/server';
+import { productDocInstallStatusSavedObjectTypeName } from '../common/consts';
+import type { ProductDocBaseConfig } from './config';
+import {
+  ProductDocBaseSetupContract,
+  ProductDocBaseStartContract,
+  ProductDocBaseSetupDependencies,
+  ProductDocBaseStartDependencies,
+  InternalRouteServices,
+} from './types';
+import { productDocInstallStatusSavedObjectType } from './saved_objects';
+import { PackageInstaller } from './services/package_installer';
+import { InferenceEndpointManager } from './services/inference_endpoint';
+import { ProductDocInstallClient } from './services/doc_install_status';
+import { SearchService } from './services/search';
+import { registerRoutes } from './routes';
+
+export class ProductDocBasePlugin
+  implements
+    Plugin<
+      ProductDocBaseSetupContract,
+      ProductDocBaseStartContract,
+      ProductDocBaseSetupDependencies,
+      ProductDocBaseStartDependencies
+    >
+{
+  private logger: Logger;
+  private routeServices?: InternalRouteServices;
+
+  constructor(private readonly context: PluginInitializerContext<ProductDocBaseConfig>) {
+    this.logger = context.logger.get();
+  }
+  setup(
+    coreSetup: CoreSetup<ProductDocBaseStartDependencies, ProductDocBaseStartContract>,
+    pluginsSetup: ProductDocBaseSetupDependencies
+  ): ProductDocBaseSetupContract {
+    coreSetup.savedObjects.registerType(productDocInstallStatusSavedObjectType);
+
+    const router = coreSetup.http.createRouter();
+    registerRoutes({
+      router,
+      getServices: () => {
+        if (!this.routeServices) {
+          throw new Error('getServices called before #start');
+        }
+        return this.routeServices;
+      },
+    });
+
+    return {};
+  }
+
+  start(
```

Comment:
I have a general concern around audit logging. We're leveraging the internal user to manage SO lifecycles, doc assets, and searching.

I think we should consider adding our own audit log entries for user-initiated events (via `core.security.audit.asScoped(request).log({...})`), so that auditors have some visibility into these operations.

Developer docs for audit logging are located at `dev_docs/key_concepts/audit_logging.mdx`.
End-user docs (`docs/user/security/audit-logging.asciidoc`) will need updating if we add new audit events.

Reply:
So, the thing is: if we decide to go with task manager for installation, then we no longer have a user request during execution of the said background task. How is audit logging usually managing that kind of use case?

Reply:
> So, the thing is: if we decide to go with task manager for installation, then we no longer have a user request during execution of the said background task. How is audit logging usually managing that kind of use case?

Good question. Most other use cases store credentials for the end user (via a granted API Key), and use those credentials when executing the background task. I think our use case is a bit different though, because these operations can be managed by either the end user or the internal user.

Maybe we can take a middle ground approach, and record an audit event that captures the intent with a reference to the task id. Something like:

1/ Use task manager to schedule a task
2/ TM returns a `taskId` once scheduled
3/ We record an audit event, with `outcome=unknown`. e.g. `User is requesting installation of product documentation for AI Assistants. Task ID=[taskId]`.

> Side question - for system tasks done on behalf of the system (not a user), such as ...
> I assume we don't want audit logging?

Correct, we do not need to audit system events. ES will capture this in their audit log, and that will be "good enough".